packages/editor/src/components/block-list/block.js

```diff
@@ -776,7 +774,25 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
 	};
 } );
 
+const getBlock = memize( ( name, attributes ) => {
+	return {
+		name,
+		attributes,
+	};
+} );
+
 export default compose(
+	// This withSelect is only necessary for backward compatibility
+	// Users of the editor.BlockListBlock filter used to be able to access the block prop
+	// Ideally these blocks would rely on the clientId prop only.
+	withSelect( ( select, { clientId } ) => {
+		const { getBlockName, getBlockAttributes } = select( 'core/editor' );
+		const name = getBlockName( clientId );
+		const attributes = getBlockAttributes( clientId );
+		return {
+			block: getBlock( name, attributes ),
+		};
+	} ),
 	withFilters( 'editor.BlockListBlock' ),
```

Comment:
Do we really need to keep `withFilters( 'editor.__experimentalBlockListBlock' )`? It looks like it can be also re-created using the public filter. I might be missing something though.

Reply:
Restored the order, this filter is useless now.